scripts/api_check.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+# This script makes sure every API header can be included in isolation
+
+for filename in $(find ../include -name *.hpp); do
+    echo Checking that $filename is self-contained
+    rm 1.cpp
+    echo "#include \"$filename\"" >> 1.cpp
+    echo "int main() {}" >> 1.cpp
+    g++ -std=c++11 1.cpp || exit 1
```

Comment:
Does it provide indication which header file is tried ?

Reply:
The last `echo Checking that $filename is self-contained` that has no `echo $filename is OK!` failed...

Reply:
Not just that, you will get the exact message from G++